src/User.js

```diff
@@ -1,11 +1,93 @@
+const recipeData = require('../data/recipes');
+const ingredientsData = require('../data/ingredients');
+
 class User {
-  constructor(name, id, pantry) {
-    this.name = name;
-    this.id = id;
-    this.pantry = pantry;
+  constructor(user) {
+    this.name = user.name;
+    this.id = user.id;
+    this.pantry = user.pantry;
+    this.favoriteRecipes = [];
+    this.toCook = [];
   };
+
+  addFavoriteRecipe(idOfClick) {
+    recipeData.find(recipe => {
+      if (recipe.id === idOfClick) {
+        this.favoriteRecipes.push(recipe);
+      }
+    });
+  }
+
+  removeFavoriteRecipe(idOfClick) {
+    let matchingRecipe = recipeData.find(recipe => recipe.id === idOfClick);
+    let indexOfMatchingRecipe = this.favoriteRecipes.indexOf(matchingRecipe);
+    this.favoriteRecipes.splice(indexOfMatchingRecipe, 1);
+  }
+
+  addRecipeToCook(idOfClick) {
+    recipeData.find(recipe => {
+      if (recipe.id === idOfClick) {
+        this.toCook.push(recipe);
+      }
+    });
+  }
+
+  filterRecipesByType(searchWord) {
+    let savedRecipes = this.favoriteRecipes.concat(this.toCook);
+    let lowerCaseSearchWord = searchWord.toLowerCase();
+
+    let foundRecipesByType =
+    savedRecipes.reduce((matchingRecipes, recipe) => {
+      recipe.tags.forEach(tag => {
+        if (tag === lowerCaseSearchWord) {
+          matchingRecipes.push(recipe);
+        }
+      });
+        return matchingRecipes;
+    }, []);
+
+    if (foundRecipesByType.length === 0) {
+      return 'We\'re Sorry, Your search did not return any results!'
+    } else {
+      return foundRecipesByType;
+    }
+  }
+
+  findRecipeByName(searchWord) {
+    let savedRecipes = this.favoriteRecipes.concat(this.toCook);
```

Comment:
bringing favorite recipes & recipes to cook arrays into one as those are the saved recipes a user will have.